metrics.md

```diff
@@ -3,44 +3,59 @@
-| beaconchain_current_finalized_epoch         | Gauge       | Current finalized epoch                                           | On each epoch                       |
-| beaconchain_current_prev_justified_epoch    | Gauge       | Current previously justified epoch                                | On each epoch                       | 
-| beaconchain_current_epoch_live_validators   | Gauge       | Number of active validators who reported for the current epoch    | On each epoch                       |
-| beaconchain_previous_epoch_live_validators  | Gauge       | Number of active validators who reported for the previous epoch   | On each epoch
-| beaconchain_reorg_events_total              | Counter     | Occurrence of a reorganization of the chain                       | On fork choice                      |
-| beaconchain_pending_deposits                | Gauge       | Number of pending deposits                                        | Upon processing eth1 events watching the deposit contract |
-| beaconchain_pending_exits                | Gauge       | Number of pending exits                                        | On each epoch                              |
-| beaconchain_total_deposits                  | Gauge       | Number of total deposits                                          | Upon processing eth1 events watching the deposit contract |
-| beaconchain_previous_epoch_stale_blocks     | Gauge       | Number of blocks not included into canonical chain in the previous epoch                                                                                                                  | On each epoch                       |
-| beaconchain_propagated_attestations         | Gauge       | Number of distinct attestations to a slot received from the wire                                                                                                                 | When attestation inclusion delay passed     |
+|-----------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `beaconchain_peers`                           | Gauge       | Tracks number of peers                                      | On peer add/drop     |
+| `beaconchain_slot`                            | Gauge       | Latest slot of the beacon chain state                       | On slot              |
+| `beaconchain_head_slot`                       | Gauge       | Slot of the head block of the beacon chain                  | On fork choice       |
+| `beaconchain_head_root`                       | Gauge       | Root of the head block of the beacon chain                  | On fork choice       |
+| `beaconchain_finalized_epoch`                 | Gauge       | Current finalized epoch                                     | On epoch transition  |
+| `beaconchain_finalized_root`                  | Gauge       | Current finalized root                                      | On epoch transition  |
```

Comment:
This isn't a scalar value, how should the root be reported to a Gauge metric?

Reply:
@prestonvanloon how do you do this conversion?

We do `root.to_low_u64_le() as i64`.

It's pretty ugly, I just picked something that's easy and works. Happy to change.

Reply:
We don't support metrics for {finalized,justified} roots. We haven't found a use case for it yet. If two nodes report the same finalized epoch, they must have the same finalized root (unless something is fundamentally wrong which would be obvious in other places). 

How does data look on a graph look for you all?

Reply:
The only two metric types I am aware of are `Gauge` and `Counter`, but `bytes` type is supported as an output so this seemed the most natural configuration. Might be missing something though

Reply:
Ah, realized that they meant "number of bytes" instead of "byte" output...

Reply:
Bytes as a unit would be referring to the size of something, not actual byte data.

Reply:
Does prometheus not have some sort of status or string type available? Digging through the docs now

Reply:
> We haven't found a use case for it yet.

Use case would be inducing >1/3 double signing and (potentially) getting the network to partition. Won't be needing it next week.

We can pull it out if not a trivial thing to report here

Reply:
You can use labels, but they are not recommended for high cardinality data.
https://prometheus.io/docs/practices/naming/#labels

I think converting the root to a scalar value as @paulhauner suggested would be the only way to support roots as metric data. I imagine this data on a graph would be pretty wild though

Reply:
> You can use labels, but they are not recommended for high cardinality data.

Labels are used for filtering a specific metric, right?
For example `beacon_validator_total` with `status="pending|active|exited|withdrawable`

Reply:
> I imagine this data on a graph would be pretty wild though

lol

Reply:
> Labels are used for filtering a specific metric, right?
For example beacon_validator_total with status="pending|active|exited|withdrawable

This would be a good example, yes. Another one we use labels for are pubsub message counts by topic.

Reply:
> How does data look on a graph look for you all?

It's not _great_ to be honest, but it does give some indication that nodes are actually on the same chain.

I thought Prysm was doing some tracking of block roots in Prom? I recall someone saying you were doing a similar root -> int cast?

Reply:
Here's what it looks like:

![grafana](https://user-images.githubusercontent.com/6660660/64230970-a62fac00-cf31-11e9-9da5-0412bf58ab56.png)

It gets a bit funky when the polling intervals don't line up exactly with the slots. Plus, it's doing some funny stuff where it's getting the average between hashes (which is not so useful, lol).

It's pretty clear that Grafana/Prom don't intend to be used like this, but on the other hand it does give a useful indication that the nodes are finalizing on the same block. I could take it or leave it for this spec.

_On a side note: it is kinda fun to watch the patterns that the hashes make and it is an interesting way to think about the chain._

Reply:
Roots sound OK to me. My only ask is that we clarify what it means to push roots as scalar values.

It sounds like @paulhauner is using the last 8 bytes, little endian, as signed 64 bit integers.